l10n_it_fatturapa_pec/models/fatturapa_attachment_out.py

```diff
@@ -31,135 +26,25 @@ def parse_pec_response(self, message_dict):
-                        'last_sdi_response': 'SdI ID: {}; '
-                        'Message ID: {}; Receipt date: {}; '
-                        'Delivery date: {}'.format(
-                            id_sdi, message_id, receipt_dt, delivery_dt)
-                    })
-                elif message_type == 'NE':  # 4A. Notifica Esito per PA
-                    esito_committente = root.find('EsitoCommittente')
-                    if esito_committente is not None:
-                        # more than one esito?
-                        esito = esito_committente.find('Esito')
-                        if esito is not None:
-                            if esito.text == 'EC01':
-                                state = 'accepted'
-                            elif esito.text == 'EC02':
-                                state = 'rejected'
-                            fatturapa_attachment_out.write({
-                                'state': state,
-                                'last_sdi_response': 'SdI ID: {}; '
-                                'Message ID: {}; Response: {}; '.format(
-                                    id_sdi, message_id, esito.text)
-                            })
-                elif message_type == 'DT':  # 5. Decorrenza Termini per PA
-                    description = root.find('Descrizione')
-                    if description is not None:
-                        fatturapa_attachment_out.write({
-                            'state': 'validated',
-                            'last_sdi_response': 'SdI ID: {}; '
-                            'Message ID: {}; Receipt date: {}; '
-                            'Description: {}'.format(
-                                id_sdi, message_id, receipt_dt,
-                                description.text)
-                        })
-                # not implemented - todo
-                elif message_type == 'AT':  # 6. Avvenuta Trasmissione per PA
-                    description = root.find('Descrizione')
-                    if description is not None:
-                        fatturapa_attachment_out.write({
-                            'state': 'accepted',
-                            'last_sdi_response': (
-                                'SdI ID: {}; Message ID: {}; Receipt date: {};'
-                                ' Description: {}'
-                            ).format(
-                                id_sdi, message_id, receipt_dt,
-                                description.text)
-                        })
+        sdi_channel_model = self.env['sdi.channel']
+        attachments = sdi_channel_model.receive_notification(
+            {
+                notification.fname: notification.content
+                for notification in notifications
+            },
+        )
 
-                message_dict['res_id'] = fatturapa_attachment_out.id
+        # Link the message to the last attachment updated
+        message_dict['res_id'] = attachments[-1].id
```

Comment:
Ma ci possono essere casi in cui `attachments` è più grande di 1?

Reply:
A guardare puramente il codice, sì.
Nella pratica poi magari no, ma potrebbe forse succedere quando invieremo degli zip e verranno quindi trovate diverse FE per una stessa notifica?

Da quel che vedo nel codice prima di questa PR, ci possono essere più allegati da gestire come notifiche SdI, e per ogni notifica potrebbe essere trovata una FE diversa.
Ho mantenuto il comportamento precedente che sovrascriveva
```
message_dict['res_id'] = fatturapa_attachment_out.id
```
per ogni allegato trovato, quindi la notifica veniva segnalata solo nell'ultimo allegato trovato.